server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -341,10 +348,17 @@ private boolean upload(long primaryTerm, long generation, long maxSeqNo) throws
                 transferSnapshotProvider,
                 new RemoteFsTranslogTransferListener(generation, primaryTerm, maxSeqNo)
             );
+        } finally {
+            syncPermit.release(SYNC_PERMIT);
```

Comment:
This is deterministic to run if `syncPermit.tryAcquire(SYNC_PERMIT) == false` does not hold true.